### ISSUE TITLE
Reconstructions Visualization of LUPerson

### ISSUE THREAD

**Issue by githubiubiu** [junkunyuan/HAP#10]
Hi! Thank you for your intriguing work. I reproduced the vit_large based on HAP, and when I visualized the reconstructed image, I found the following:

The vis demo is from MAE repo, and I am confused by the highlighted color blocks in the reconstructed image. I'm not sure if it's a problem with the vis script or the pretrained model, but the same problem also exists when using the vit_b pretrained model in HAP repo.

**Comment by githubiubiu** [comment]
![demo](https://github.com/junkunyuan/HAP/assets/52188268/1dddbf12-55b0-4303-83a2-2fe5e2b0a644)

**Comment by junkunyuan** [comment]
> Hi! Thank you for your intriguing work. I reproduced the vit_large based on HAP, and when I visualized the reconstructed image, I found the following:
> 
> The vis demo is from MAE repo, and I am confused by the highlighted color blocks in the reconstructed image. I'm not sure if it's a problem with the vis script or the pretrained model, but the same problem also exists when using the vit_b pretrained model in HAP repo.

This is an intriguing question that I have come across previously.

The reason behind this lies in the fact that the version employed for inference utilizes normalized pixels as the prediction target during pre-training, whereas the version used for visualization employs unnormalized pixels as the prediction target.

Therefore, if you wish to visualize the results, you may need to train a new version that employs unnormalized pixels as the prediction target.

You can find a detailed answer from Kaiming by following this link:

https://github.com/facebookresearch/mae/issues/12#issuecomment-1011689674
